File: dagger2/src/main/java/pgreze/dagger2/ui/hello/HelloPresenter.java
package pgreze.dagger2.ui.hello;

import javax.inject.Inject;

import pgreze.dagger2.analytics.AnalyticsManager;
import pgreze.dagger2.di.UILifecycleScope;

@UILifecycleScope
public class HelloPresenter {

    private final AnalyticsManager analyticsManager;
    private HelloFragment fragment;

    private int nClick = 0;

    @Inject
    public HelloPresenter(AnalyticsManager analyticsManager) {
        this.analyticsManager = analyticsManager;
    }

    public void init(HelloFragment fragment) {
        this.fragment = fragment;
    }

    public void onHelloClick() {
        nClick++;
        fragment.getHelloBtn().setText(String.format("Retry (%s time(s))", nClick));
        analyticsManager.event("Hello Button Clicked");
    }
}

File: dagger/android/src/main/kotlin/pgreze/dagger2/ui/hello/HelloPresenter.kt
package pgreze.dagger2.ui.hello

import javax.inject.Inject

import pgreze.dagger2.analytics.AnalyticsManager
import pgreze.dagger2.di.AndroidLifecycleScope

@AndroidLifecycleScope
class HelloPresenter @Inject constructor(
        private val analyticsManager: AnalyticsManager
) {
    private var fragment: HelloFragment? = null

    private var nClick = 0

    fun init(fragment: HelloFragment) {
        this.fragment = fragment
    }

    fun onHelloClick() {
        nClick++
        fragment?.helloText = String.format("Retry (%s time(s))", nClick)
        analyticsManager.event("Hello Button Clicked")
    }
}
